### ISSUE TITLE
Applications/windows open on wrong ttys

### ISSUE THREAD

**Issue by sebastian-burlacu** [swaywm/sway#4274]
Debug info follows brief problem description.

I reboot my machine and log in to default Ubuntu. I press Ctrl + Alt + F3 to get to a new tty and I log in, then I run 'sway'. If it complains about permissions I 'sudo chmod a+s /usr/local/bin/sway' and then launch sway. It opens fine; no error messages. However when I press Super + Return, nothing happens. I switch back to my default tty with gnome DE and then my terminals pop up there. Within gnome, I launch, say, my file navigator (Nautilus) and it doesn't open. I switch back to Sway and there it is.

This behaviour persists across reboots. Why are commands from one tty launching applications into another one?

If you are using the nvidia proprietary driver for any reason, you have two choices:

1. Uninstall it and use nouveau instead
2. Use X11+i3 and close your browser tab

If `lsmod | grep nvidia | wc -l` shows anything other than zero, your bug report is not welcome here.

Additionally, problems with Firefox are almost certainly Firefox bugs, not sway bugs. Start by submitting your issue to the Firefox bugzilla and come back here only after they confirm otherwise.

Otherwise, please include the following four components in your bug report: sway version, debug log, configuration (if applicable), and an explanation of steps taken to reproduce the issue. If sway crashes, also include a stack trace.

Obtain your version like so:

    swaymsg -t get_version

If this doesn't work, use:

    sway -v

* Sway Version: sway version 1.1-rc1-46-gd0d01810 (Jun 21 2019, branch 'master')

Obtain a debug log like so:

    sway -d 2> ~/sway.log

This will record information about sway's activity when it's running. Briefly reproduce your problem and exit sway.  When preparing a debug log, brevity is important - start up sway, do the minimum work necessary to reproduce the error, then close sway.

Upload the debug log to a pastebin service such as [gist.github.com](https://gist.github.com), and link to it below.

* Debug Log: https://gist.github.com/sebastian-burlacu/a36cf695d3dfc1d85d6f050dcef53ee5

You should try to reproduce the issue with the default configuration. If you cannot, please reproduce with a minimal configuration, upload the config to a pastebin service, and link to it below.

* Configuration File: <default file>

Finally, explain the steps you took in plain English to reproduce the problem below.

* Stack Trace, if sway crashes:

It doesn't crash; it just doesn't work properly.

If you use systemd, you should be able to open the coredump of the most recent crash with GDB like so:

    coredumpctl gdb sway

And then type `bt full` to obtain the stack trace.

**Comment by ascent12** [comment]
Does this specifically happen with gnome programs? Because they don't respect WAYLAND_DISPLAY, completely fall over when using multiple TTYs, and this pisses me off to no end.
If it is, then it's a Gnome issue and there is nothing we can do about it.

**Comment by sebastian-burlacu** [comment]
This happens with any app, but specifically I noticed it happening with urxvt terminal (since that's what the Super + Return shortcut is launching). I tried pressing it a few times, then I gave up waiting and switched back to Gnome, and then had several urxvt windows pop up.

**Comment by arkhan** [comment]
I have the same issue!

**Comment by Isti115** [comment]
For me, if there is a sway session open, some applications (I currently remember Termite, MuseScore and Thunar) completely refuse to open under Xorg, they always start in the tty running sway.

**Comment by emersion** [comment]
This is a GTK bug.

**Comment by Isti115** [comment]
If I have a sway session open, there are some apps that refuse to open under Xorg. No matter how I try to launch for example Termite, Thunar or MuseScore, they land on the tty that runs sway.
